server/sonar-web/src/main/js/apps/projectActivity/components/forms/AddGraphMetric.js

```diff
@@ -71,111 +140,99 @@ export default class AddGraphMetric extends React.PureComponent {
         return true;
       })
       .map((metric /*: Metric */) => ({
-        value: metric.key,
+        key: metric.key,
         label: getLocalizedMetricName(metric)
       }));
   };
 
-  openForm = () => {
-    this.setState({
-      open: true
+  getSelectedMetricsElements = (
+    metrics /*: Array<Metric> */,
+    selectedMetrics /*: Array<string> | null */
+  ) => {
+    const selected /*: Array<string> */ =
+      selectedMetrics === null ? this.props.selectedMetrics : selectedMetrics;
+    return metrics.filter(metric => selected.includes(metric.key)).map((metric /*: Metric */) => ({
+      key: metric.key,
+      label: getLocalizedMetricName(metric)
+    }));
+  };
+
+  toggleForm = () => {
+    this.setState(state => {
+      return { open: !state.open };
     });
   };
 
-  closeForm = () => {
-    this.setState({
-      open: false,
-      selectedMetric: undefined
+  onSearch = (query /*: string */) => {
+    return new Promise(resolve => {
+      this.setState({ query }, () => {
+        resolve();
```

Comment:
there is no point to wait the state update to resolve the promise, you can simply do:
```javascript
this.setState({ query });
return Promise.resolve();
```